src/js/common/zoom.js

```diff
@@ -0,0 +1,71 @@
+{
+
+const filter_in_range_data = (args, range) => {
+  const is_data_in_range = (data, range) => {
+    return data > Math.min(range[0], range[1]) && data < Math.max(range[0], range[1]);
+  }
+
+  return d => {
+    const x_in_range = !('x' in range) || is_data_in_range(d[args.x_accessor], range.x);
+    const y_in_range = !('y' in range) || is_data_in_range(d[args.y_accessor], range.y);
+    return x_in_range && y_in_range;
+  }
+}
+
+const zoom_to_data_domain = (args, range) => {
+  const raw_data = args.raw_data || args.data;
+  if (!('raw_data' in args)) {
+    args.raw_domain = {
```

Comment:
I would probably put `raw_domain` and `raw_data` under `args.processed`, since they don't represent user input.

Reply:
I think so, but there are many variables under args.